src/api/python/wheels/build_wheel.py

```diff
@@ -0,0 +1,155 @@
+# in the top-level source directory and their institutional affiliations.
+# All rights reserved.  See the file COPYING in the top-level source
+# directory for licensing information.
+# ############################################################################
+#
+# Script for building wheels distribution for pycvc5
+#
+# Example usage (from top-level directory):
+#   python3 ./src/api/python/wheels/build_wheel.py bdist_wheel
+# Configures and builds in directory ./build
+# Creates wheel in ./dist
+#
+##
+
+import os
+import re
+import sys
+import platform
+import subprocess
+import multiprocessing
+import shutil
+import glob
+
+from setuptools import setup, Extension
+from setuptools.command.build_ext import build_ext
+from distutils.version import LooseVersion
+
+def get_project_src_path():
+    # expecting this script to be in src/api/python/wheels
+    # The project source directory is 4 directories up
+    # (5 dirnames because start with file)
+    project_src_path = __file__
+    for i in range(5):
+        project_src_path = os.path.dirname(project_src_path)
+    return os.path.abspath(project_src_path)
+
+
+def get_cvc5_version():
+    project_src_path = get_project_src_path()
+
+    # read CMakeLists.txt to get version number
+    version = dict()
+    str_pattern = 'set\(CVC5_(?P<component>MAJOR|MINOR|RELEASE)\s*(?P<version>\d+)\)'
+    pattern = re.compile(str_pattern)
+    with open(project_src_path + '/CMakeLists.txt', 'r') as f:
+        for line in f:
+            line = line.strip()
+            m = pattern.search(line)
+            if m:
+                gd = m.groupdict()
+                version[gd['component']] = gd['version']
+                if len(version) == 3:
+                    break
+
+    assert len(version) == 3, 'Expecting MAJOR, MINOR and RELEASE'
+    return version['MAJOR'], version['MINOR'], version['RELEASE']
+
+
+class CMakeExtension(Extension):
+    def __init__(self, name, sourcedir=''):
+        Extension.__init__(self, name, sources=[])
+        self.sourcedir = os.path.abspath(sourcedir)
+
+
+class CMakeBuild(build_ext):
+    def run(self):
+        try:
+            out = subprocess.check_output(['cmake', '--version'])
+        except OSError:
+            raise RuntimeError(
+                "CMake must be installed to build the following extensions: " +
+                ", ".join(e.name for e in self.extensions))
+
+        if self.is_windows():
+            cmake_version = LooseVersion(
+                re.search(r'version\s*([\d.]+)', out.decode()).group(1))
+            if cmake_version < '3.1.0':
+                raise RuntimeError("CMake >= 3.1.0 is required on Windows")
+
+        for ext in self.extensions:
+            self.build_extension(ext)
+
+    @staticmethod
+    def is_windows():
+        tag = platform.system().lower()
+        return tag == "windows"
+
+    @staticmethod
+    def is_linux():
+        tag = platform.system().lower()
+        return tag == "linux"
```

Comment:
Seems to be unused.

Reply:
Good catch, addressed in a1f6c9c540e12bdca91526ea28e94ed2eaf545d9.